ring.go

```diff
@@ -273,7 +274,15 @@ func (sig *RingSig) Verify(m [32]byte) bool {
 // Link returns true if the two signatures were created by the same signer,
 // false otherwise.
 func Link(sigA, sigB *RingSig) bool {
-	return sigA.image.Equals(sigB.image)
+	switch sigA.Ring().curve.(type) {
+	case *ed25519.CurveImpl:
```

Comment:
OPTIONAL

-----------


I don't understand the foundational cryptography well enough to comment here but I think we could #PUC (Please Update Comment) for future devs.

```go
# - Ed25519 has a cofactor of 8
# - If one of the 4 small subgroups is added to the image, linkability can be broken
# - Multiplying both images by the cofactor reduces the added subgroups to the identity element
# Ref: https://monero.stackexchange.com/questions/8671/what-are-the-hex-representations-of-the-small-subgroup-curve-points-on-ed25519
```

Did some ChatGPTing to understand: https://chatgpt.com/share/a82163ea-3f52-4767-8602-b18d46e59b3b

😅